src/lib.rs

```diff
@@ -123,15 +134,26 @@ impl Contract {
         self.posts.push(&post.into());
         self.post_to_parent.insert(&id, &parent_id);
 
-        let mut siblings = self
-            .post_to_children
-            .get(&parent_id)
-            .unwrap_or_else(|| panic!("Parent id {} not found", parent_id));
+        let mut siblings = self.post_to_children.get(&parent_id).unwrap_or_else(
+            || Vec::new(), // panic!("Parent id {} not found", parent_id)
+        );
         siblings.push(id);
         self.post_to_children.insert(&parent_id, &siblings);
 
         // Don't forget to add an empty list of your own children.
         self.post_to_children.insert(&id, &vec![]);
+
+        if parent_id != ROOT_POST_ID {
+            let parent_post: Post = self
+                .posts
+                .get(parent_id)
+                .unwrap_or_else(|| panic!("Parent post with id {} not found", parent_id))
+                .into();
+            let parent_author = parent_post.author_id;
+            Some(notify::notify_reply(parent_id, parent_author))
```

Comment:
I am not sure we need to wait for `notify::notify_reply`, which is what would happen if we return `Promise` instead of just dropping it upon creation.

Reply:
That's good to know! Not waiting for notify is better, so I'll change to drop after creation.